constraint/pkg/externaldata/cache.go

```diff
@@ -0,0 +1,51 @@
+package externaldata
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/apis/externaldata/v1alpha1"
+)
+
+type ProviderCache struct {
+	cache map[string]v1alpha1.Provider
+	mux   sync.RWMutex
+}
+
+func NewCache() *ProviderCache {
+	return &ProviderCache{
+		cache: make(map[string]v1alpha1.Provider),
+	}
+}
+
+func (c *ProviderCache) Get(key string) (v1alpha1.Provider, error) {
+	if v, ok := c.cache[key]; ok {
+		return v, nil
```

Comment:
Should we return a deepcopy to prevent callers from corrupting the cache?

Reply:
updated